protocol/messenger_handler.go

```diff
@@ -1620,6 +1620,13 @@ func (m *Messenger) HandleCommunityRequestToJoinResponse(state *ReceivedMessageS
 		if err != nil {
 			return err
 		}
+
+		// we merge to include chats in response signal to joining a community
+		err = state.Response.Merge(response)
+		if err != nil {
+			return err
+		}
+
 		if len(response.Communities()) > 0 {
 			communitySettings := response.CommunitiesSettings()[0]
```

Comment:
can we extend `MessengerResponse` to also merge `communitySettings` so we can get rid of these lines?

Reply:
ah sorry forgot it, we had discussed this as well, I will take care of this :)

Reply:
Done